packages/lib/server/queries/teams/index.ts

```diff
@@ -61,115 +64,100 @@ export async function getTeamWithMembers(args: {
-        ...baseEventTypeSelect,
       },
-    },
-    inviteTokens: {
-      select: {
-        token: true,
-        expires: true,
-        expiresInDays: true,
-        identifier: true,
+      inviteTokens: {
+        select: {
+          token: true,
+          expires: true,
+          expiresInDays: true,
+          identifier: true,
+        },
       },
     },
-  });
+  } as const;
 
-  const where: Prisma.TeamFindFirstArgs["where"] = {};
+  const teamOrOrg = isOrgView ? await getOrg(arg) : await getTeam(arg);
 
-  if (userId) where.members = { some: { userId } };
-  if (orgSlug && orgSlug !== slug) {
-    where.parent = getSlugOrRequestedSlug(orgSlug);
-  }
-  if (id) where.id = id;
-  if (slug) where.slug = slug;
-  if (isOrgView) {
-    // We must fetch only the organization here.
-    // Note that an organization and a team that doesn't belong to an organization, both have parentId null
-    // If the organization has null slug(but requestedSlug is 'test') and the team also has slug 'test', we can't distinguish them without explicitly checking the metadata.isOrganization
-    // Note that, this isn't possible now to have same requestedSlug as the slug of a team not part of an organization. This is legacy teams handling mostly. But it is still safer to be sure that you are fetching an Organization only in case of isOrgView
-    where.metadata = {
-      path: ["isOrganization"],
-      equals: true,
-    };
-  }
+  if (!teamOrOrg) return null;
 
-  const teams = await prisma.team.findMany({
-    where,
-    select: teamSelect,
-  });
-
-  if (teams.length > 1) {
-    logger.error("Found more than one team/Org. We should be doing something wrong.", {
-      where,
-      teams: teams.map((team) => ({ id: team.id, slug: team.slug })),
-    });
-  }
-
-  const team = teams[0];
-  if (!team) return null;
-
-  // This should improve performance saving already app data found.
-  const appDataMap = new Map();
-  const members = team.members.map((obj) => {
-    const { credentials, ...restUser } = obj.user;
+  const members = teamOrOrg.members.map((m) => {
+    const { credentials, ...restUser } = m.user;
     return {
       ...restUser,
-      role: obj.role,
-      accepted: obj.accepted,
-      disableImpersonation: obj.disableImpersonation,
+      role: m.role,
+      accepted: m.accepted,
+      disableImpersonation: m.disableImpersonation,
       subteams: orgSlug
-        ? obj.user.teams.filter((obj) => obj.team.slug !== orgSlug).map((obj) => obj.team.slug)
+        ? m.user.teams
+            .filter((membership) => membership.team.slug !== orgSlug)
+            .map((membership) => membership.team.slug)
         : null,
-      avatar: `${WEBAPP_URL}/${obj.user.username}/avatar.png`,
-      orgOrigin: getOrgFullOrigin(obj.user.organization?.slug || ""),
+      avatar: `${WEBAPP_URL}/${m.user.username}/avatar.png`,
+      orgOrigin: getOrgFullOrigin(m.user.organization?.slug || ""),
       connectedApps: !isTeamView
```

Comment:
What is the use of isTeamView ? Do we need both variable isOrgView and isTeamView

Reply:
`isTeamView` is actually meant to decide if credentials are supposed to be available in the function return value or not.

It is true on team booking pages. The name is quite confusing though, I fully agree